### DISCUSSION TITLE
Unable to upload files even after following https://www.librechat.ai/docs/configuration/rag_api

### DISCUSSION THREAD

**Discussion by ScubaSlavvo** [danny-avila/LibreChat#4727] [Help Wanted]
Hi all,
I a using the OpenAI api and wanted to upload files. I followed the steps on the page https://www.librechat.ai/docs/configuration/rag_api
added a
 #. RAG_API_URL=http://host.docker.internal:8000 (without the "." i just didn't want the # to edit the text)

line under the "rag" section in the env.example file, along with 
sk-your-openai-api-key-example under RAG_OPENAI_API_KEY (also tried just pasting my openAI API key but it did not work)
The error I am getting is : 2024-11-14 14:43:43,926 - root - ERROR - Error code: 401 - {'error': {'message': 'Incorrect API key provided: user_pro*ided. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}
Other than this, I can use the API normally.

I am a total beginner at this so i might have messed something basic up. I would appreciate any tips or help ! 
Thanks in advance !!

**Comment by fuegovic** [comment]
The `RAG_OPENAI_API_KEY` field requires your OpenAI API key. Using `sk-your-openai-api-key-example` will not work and will result in an 'invalid_api_key' error.

You might consider setting your OpenAI API key higher in the .env file, rather than using "user_provided", at this specific line: https://github.com/danny-avila/LibreChat/blob/493e64e6dbe0c362a51415486806ae8483cd653e/.env.example#L174.

**Reply by fuegovic** [reply]
Also, avoid placing a `#` before `RAG_OPENAI_API_KEY=...` in the .env file, as this will cause the value to be commented out of the configuration.

**Reply by ScubaSlavvo** [reply]
Thanks for your answer!
I modified the things you mentioned, here is what it looks like now : 
![image](https://github.com/user-attachments/assets/5ca0e325-5f4d-4724-a1e4-eedac5d004a8)
![image](https://github.com/user-attachments/assets/32bb238a-1ba8-4dec-a127-0988092a26b8)

However, I am still getting : 
2024-11-15 08:10:58 2024-11-15 07:10:58,944 - root - ERROR - Error code: 401 - {'error': {'message': 'Incorrect API key provided: user_pro*ided. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}

I am using the same API key as the one i put into the API which allowed me to use it so I know it is correct. Could the issue be anything else ? Are these the only things I need to do for the rag to work ? I might have forgotten or skipped over something else
